File: problems/1292. Maximum Side Length of a Square with Sum Less than or Equal to Threshold/solution.rs
impl Solution {
    pub fn max_side_length(mat: Vec<Vec<i32>>, threshold: i32) -> i32 {
        let mut side: i32 = 1;
        let mut max_side: i32 = 0;
        let mut i: i32 = 0;
        let mut j: i32 = 0;
        let i_len = mat.len();
        let j_len = mat[i as usize].len();
        while i + side <= i_len as i32 && j + side <= j_len as i32 {
            let mut sum = 0;
            for k in (i..i+side) {
                for l in (j..j+side) {
                    sum += mat[k as usize][l as usize];
                }
            }
            if sum <= threshold {
                max_side = side;
                side += 1;
                continue
            }
            if j+side >= j_len as i32{
                j = 0;
                i +=1;
            } else {
                j +=1;
            }
        }
        return max_side
    }
}

File: problems/1292. Maximum Side Length of a Square with Sum Less than or Equal to Threshold/solution.py
class Solution:
    def maxSideLength(self, mat: List[List[int]], threshold: int) -> int:
        side = 1
        max_side = 0
        i = 0
        j = 0
        i_len = len(mat)
        j_len = len(mat[i])
        while i + side <= i_len and j + side <= j_len:
            sum = 0
            for k in range(i, i + side):
                for l in range(j, j + side):
                    sum += mat[k][l]
            if sum <= threshold:
                max_side = side
                side += 1
                continue
            if j + side >= j_len:
                j = 0
                i += 1
            else:
                j += 1
        return max_side